src/tests/ftest/deployment/critical_integration.py

```diff
@@ -0,0 +1,108 @@
+                                  " 'echo hello'".format(host))
+            command_for_inter_node = ("clush --nostdin -S -w {}"
+                                      " 'echo hello'".format(','.join(self.hostlist_servers)))
+            try:
+                out = run_command(daos_server_cmd)
+                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                if check_remote_root_access:
+                    run_command(remote_root_access)
+                IorTestBase._execute_command(self, command_for_inter_node, hosts=[host])
+            except (DaosTestError, CommandFailure) as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("Ssh check failed on following nodes.\n {}".format(failed_nodes))
+
+        for host in self.hostlist_clients:
+            dmg_version_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'dmg version -i'".format(host))
+
+            try:
+                out = run_command(dmg_version_cmd)
+                dmg_version_list.append(out.stdout.split(b' ')[2])
+            except DaosTestError as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("SSH check for client nodes failed.\n {}".format(failed_nodes))
+
+        result_daos_server = (daos_server_version_list.count(daos_server_version_list[0])
+                              == len(daos_server_version_list))
+        result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
+        result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
+
+        if (result_daos_server and result_dmg and result_client_server):
+            self.log.info("All servers have same daos version")
+            self.log.info("All clients have same daos version")
+            self.log.info("Servers and Clients have same daos version")
+        else:
+            self.log.info("Not all servers and clients have same daos version")
+            self.log.info("Clients: %s", dmg_version_list)
+            self.log.info("servers: %s", daos_server_version_list)
+            self.fail()
+
+    def test_ras(self):
+        """
+        Test Description: Verify RAS event on all server nodes from testrunner.
+                          Verify network scan and storage scan for server nodes.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=criticalintegration,ras
+        """
+
+        dmg = self.get_dmg_command()
+        rank_list = self.server_managers[0].get_host_ranks(self.hostlist_servers)
+        print("rank_list: {}".format(rank_list))
+
+        for rank in rank_list:
+            dmg.system_stop(ranks=rank)
+            time.sleep(5)
+            dmg.system_start(ranks=rank)
+            time.sleep(5)
+
+        dmg.storage_scan()
```

Comment:
Don't we need to check the results output for any errors? Is this ok just to print the output?

Reply:
+1

Reply:
Yes, idea here is to just try and run storage/network scan and verify it works on these nodes. In order to verify if storage/network scan outputs the right values it is already being done as part of our funtional testing in CI. Here we are only trying to check if the nodes are working in tandem with our daos software (hence, critical integration).